gobblin-core/src/main/java/gobblin/writer/http/BatchAndSendSyncHttpWriter.java

```diff
@@ -0,0 +1,86 @@
+package gobblin.writer.http;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import com.google.common.collect.Sets;
+
+import gobblin.http.BatchRequestBuilder;
+
+
+/**
+ * Synchronous HTTP writer which buffers and sends a batch at a time
+ *
+ * @param <D> type of record
+ * @param <RQ> type of request
+ * @param <RP> type of response
+ */
+public class BatchAndSendSyncHttpWriter<D, RQ, RP> extends HttpWriterBase<D, RQ, RP> {
+  BatchRequestBuilder<D, RQ> batchRequestBuilder;
+  protected final Collection<D> recordSet;
+  protected final int maxBatchSize;
+  protected long numRecordsWritten = 0L;
+
+  public BatchAndSendSyncHttpWriter(BatchHttpWriterBaseBuilder builder) {
```

Comment:
For the builder pattern, it doesn't have a rule that a build shouldn't have a getter. A builder is intended to prepare necessary parts for building a complex object. The contract between a builder and an instance is apparent, so it is normal to expose its detail.